### PR TITLE
doc improve datatype subsection

### PR DESCRIPTION
**Pull request by bruno-at-bareos**
This PR is an improvement of data-types subsubsection of the documentation.

    - Update all data-type to uppercase
    - Update all index to match data-type
    - Update all index syntax
    - Sort alphabetically all data-type
    - Make indentation (3) uniform everywhere
    - Add description to each data-type

#### Please check

- [X] Short description and the purpose of this PR is present _above this paragraph_
- [X] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)

##### General

- [x] PR name is meaningful
- [x] Purpose of the PR is understood
- [x] Separate commit for this PR in the CHANGELOG.md, PR number referenced is same
- [x] Commit descriptions are understandable and well formatted

##### Source code quality

- [x] Source code changes are understandable
- [x] Variable and function names are meaningful
- [x] Code comments are correct (logically and spelling)
- [x] Required documentation changes are present and part of the PR
- [x] `bareos-check-sources --since-merge` does not report any problems
- [x] `git status` should not report modifications in the source tree after building and testing

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ and since Bareos version 20 this project adheres to [Semantic Versioning](https:
 - when setting an IPv6 address using the `[DIR|SD|FD] Addresses` directive, now bareos only listens on IPv6 instead of both IPv4 and IPv6 on dual-stack. If you were using the `[DIR|SD|FD] Addresses` directive to create a dual-stack socket that would listen on both IPv6 AND IPv4, it will not work that way anymore. You should now also explicitly specify the IPv4 address in the directive [PR #882]
 
 ### Fixed
+- docs: Improve data-type Address, Addresses in CustomizingTheConfiguration [PR #968]
```
Comment (arogge) [inline CHANGELOG.md:14]: maybe that was "changed" and not "fixed", but your decision.

**Review by arogge — APPROVED** [review]
Definitely a huge improvement of that section.

**Review by arogge — CHANGES_REQUESTED** [review]
* the CHANGELOG.md record should have its own Commit
* there's a typo "complete remork" in one of the commits

File: docs/manuals/source/Configuration/CustomizingTheConfiguration.rst [hunk 2]
```diff
@@ -450,10 +451,10 @@ Data Types
 
 When parsing the resource directives, Bareos classifies the data according to the types listed below.
 
-.. _DataTypeAcl:
+.. _DataTypeACL:
 
-acl
-   :index:`\ <single: Data Type; acl>`\
+ACL
+   :index:`\ <single: Data Type; ACL>`\
```
Comment (joergsteffens) [inline docs/manuals/source/Configuration/CustomizingTheConfiguration.rst:457]:
```suggestion
   .. index::
      single: Data Type; ACL
```
An index is better written in this format. The other format has been introduced by the automatic conversion from Latex and is only required for an inline index.
If you adapt this section, best use this format, see https://docs.bareos.org/DocumentationStyleGuide/RestOverview.html#index

File: CHANGELOG.md (hunk 1 above)
Comment (bruno-at-bareos) [reply CHANGELOG.md:14]: Once upon a time it was fixed (Addresses example) But it has mutated into changed so yes better to move that.

**Review by arogge — APPROVED** [review]

File: docs/manuals/source/Configuration/CustomizingTheConfiguration.rst [hunk 3]
```diff
@@ -519,611 +521,706 @@ acl
...
 
-  :index:`\ <single: Data Type; REPLACE_OPTION>`\
+.. _DataTypeSTRINGLIST:
 
-  Description of REPLACE_OPTION
+STRINGLIST
+   .. index::
+      single: Data Type; STRINGLIST
 
+   Multiple strings, specified in multiple directives, or in a single directive, separated by commas (**,**).
```
Comment (arogge) [inline docs/manuals/source/Configuration/CustomizingTheConfiguration.rst:1165]: Isn't this a duplicate of above's `STRING_LIST`?

File: docs/manuals/source/Configuration/CustomizingTheConfiguration.rst [hunk 4]
```diff
@@ -519,611 +521,706 @@ acl
...
 
-  Description of PLUGIN_NAMES
+.. _DataTypeSPEED:
 
+SPEED
+   .. index::
+      single: Data Type; SPEED
 
-.. _DataTypePOOLTYPE:
+   The speed parameter can be specified as k/s, kb/s, m/s or mb/s.
```
Comment (arogge) [inline docs/manuals/source/Configuration/CustomizingTheConfiguration.rst:1134]: maybe we could mention what this is  k/s, kb/s, m/s and mb/s actually is?

File: docs/manuals/source/Configuration/CustomizingTheConfiguration.rst [hunk 5]
```diff
@@ -519,611 +521,706 @@ acl
...
-speed
-   :index:`\ <single: Data Type; speed>`\
+   .. code-block:: bareosconfig
 
-   The speed parameter can be specified as k/s, kb/s, m/s or mb/s.
+       Audit Events = label
+       Audit Events = restore
 
-   Don’t use quotes around the parameter, see :ref:`section-Quotes`.
+   Based on the type :ref:`string-list <DataTypeStringList>`.
```
Comment (arogge) [inline docs/manuals/source/Configuration/CustomizingTheConfiguration.rst:643]:
```suggestion
   Based on the type :ref:`STRING_LIST <DataTypeStringList>`.
```

**Review by arogge — CHANGES_REQUESTED** [review]

**Review by arogge — APPROVED** [review]
nothing left to complain about :)

**Comment by bruno-at-bareos** [comment]
I've done a final round of check and modification following Jörg recommendations.

**Review by joergsteffens — APPROVED** [review]

**Comment by bruno-at-bareos** [comment]
@joergsteffens @arogge can one of you push the merge button ?

**Review by arogge — APPROVED** [review]

**Comment by joergsteffens** [comment]
Replaced by PR #1008

**Pull request closed without merging**